bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/metadata/SiemensHvacMetadataRegistryImpl.java

```diff
@@ -0,0 +1,1259 @@
+                                        userEdit = header;
+                                    }
+                                    idxCell++;
+                                }
+
+                                if ("".equals(userName)) {
+                                    continue;
+                                }
+
+                                Pattern pattern4 = Pattern.compile("userid=(.+?)");
+                                Matcher matcher4 = pattern4.matcher(userEdit);
+
+                                SiemensHvacMetadataUser user = new SiemensHvacMetadataUser();
+                                user.setName(userName);
+
+                                if (matcher4.find()) {
+                                    userId = matcher4.group(1);
+                                    user.setId(Integer.parseInt(userId));
+                                } else {
+                                    userId = null;
+                                    user.setId(-1);
+                                }
+
+                                request = "main.app?section=settings&subsection=user&action=modify";
+                                if (userId != null) {
+                                    request = request + "&userid=" + userId;
+                                }
+                                response = lcHvacConnector.doBasicRequest(request);
+
+                                Pattern pattern5 = Pattern.compile(
+                                        "<select name=\\\"language\\\".*>((.*|\\n)*?)</select>", Pattern.MULTILINE);
+                                Matcher matcher5 = pattern5.matcher(response);
+
+                                if (matcher5.find()) {
+                                    String optionsList = matcher5.group(1);
+
+                                    Pattern pattern6 = java.util.regex.Pattern.compile(
+                                            "<option value=\\\"([^ ]*)\\\"(.*)>(.*)</option>", Pattern.MULTILINE);
+                                    Matcher matcher6 = pattern6.matcher(optionsList);
+
+                                    while (matcher6.find()) {
+                                        String id = matcher6.group(1);
+                                        String opt = matcher6.group(2);
+                                        String lang = matcher6.group(3);
+
+                                        if (opt.indexOf("selected") >= 0) {
+                                            user.setLanguage(lang);
+                                            user.setLanguageId(Integer.parseInt(id));
+                                        }
+                                    }
+                                }
+
+                                userList.put(userName, user);
+                            }
+
+                            idx++;
+
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
```

Comment:
You can either wrap the exception from `doBasicRequest`:
```suggestion
        } catch (SiemensHvacException e) {
```

Or not catch it at all, letting the original exception be thrown to the caller of this method.

I don't think there are any other exceptions to handle here?

Reply:
Yes, you're right.
I remove the try/catch and fix the throw clause on doBasicRequest.